examples/oneapi/cpp/source/table/heterogen_table.cpp

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <memory>
+#include <iostream>
+
+#include "example_util/utils.hpp"
+
+#include "oneapi/dal/table/common.hpp"
+#include "oneapi/dal/chunked_array.hpp"
+#include "oneapi/dal/table/heterogen.hpp"
+
+#include "oneapi/dal/table/row_accessor.hpp"
+
+namespace dal = oneapi::dal;
+
+template <typename Type = float>
+dal::array<Type> get_arange(std::int64_t count, std::int64_t first = 0l, std::int64_t step = 1l) {
+    auto* const raw_data = new Type[count];
+
+    for (std::int64_t i = 0l; i < count; ++i) {
+        std::int64_t value = step * i + first;
+        raw_data[i] = static_cast<Type>(value);
+    }
+
+    // Let's create an array using raw pointer and deleter
+    return dal::array<Type>(raw_data,
+                            count, //
+                            [](Type* const ptr) -> void {
+                                delete[] ptr;
+                            });
+}
+
+template <typename Type = float>
+dal::chunked_array<Type> get_chunked_arange(std::int64_t count, std::int64_t chunk_count = 2l) {
+    dal::chunked_array<Type> chunked_array(chunk_count);
+
+    std::int64_t min_count = count / chunk_count;
+    for (std::int64_t i = 0l; i != chunk_count; ++i) {
+        std::int64_t first = i * min_count;
+        std::int64_t local_count = (i + 1 == chunk_count) ? (count - first) : min_count;
+        auto chunk = get_arange<Type>(local_count, first);
+        chunked_array.set_chunk(i, chunk);
+    }
+
+    return chunked_array;
+}
+
+int main(int argc, char** argv) {
+    constexpr std::int64_t row_count = 24;
+
```

Comment:
Since these are meant to be user facing, I'd recommend adding comments to explain any potential non-intuitive parts. For example, here a comment saying "Create separate chunks of different data types" or some description describing both what "get_chunked_arange" does and about the different types.

Reply:
I have added more comments